### PR TITLE
Fix to unparse the plan with multiple UNION statements into an SQL string

### PR DESCRIPTION
**Pull request by goldmedal**
## Which issue does this PR close?

No corrsponding issue.

## Rationale for this change
Consider the following SQL:
```SQL
WITH 
    w1 AS (SELECT 'a' as col),
    w2 AS (SELECT 'b' as col),
    w3 as (SELECT 'c' as col)
SELECT * FROM w1 UNION ALL SELECT * FROM w2 UNION ALL SELECT * FROM w3
```
If a plan contains more than two UNION statements, it cannot be properly unparsed back into an SQL string.

## What changes are included in this PR?
Fix the unparsing behavior for UNION.

## Are these changes tested?
yes

## Are there any user-facing changes?
no

### CODE REVIEW

File: datafusion/sql/src/unparser/plan.rs [hunk 1]
```diff
@@ -542,10 +542,10 @@ impl Unparser<'_> {
                     right: Box::new(input_exprs[1].clone()),
                 };
 
-                query
-                    .as_mut()
-                    .expect("to have a query builder")
-                    .body(Box::new(union_expr));
+                let Some(query) = query.as_mut() else {
+                    return internal_err!("to have a query builder");
```
Comment (phillipleblanc) [inline datafusion/sql/src/unparser/plan.rs:546]:
```suggestion
                    return internal_err!("UNION ALL operator only valid in a statement context");
```

**Review by phillipleblanc — APPROVED** [review]
Thanks @goldmedal! One minor comment on the error message, but otherwise looks good to me! I should have tested more than 2 UNIONs initially 😅

File: datafusion/sql/src/unparser/plan.rs (hunk 1 above)
Comment (goldmedal) [reply datafusion/sql/src/unparser/plan.rs]: looks great! Thanks!

**Review by phillipleblanc — APPROVED** [review]

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @goldmedal  and @phillipleblanc  for the review

**Comment by goldmedal** [comment]
Thanks @alamb @phillipleblanc !

**Pull request merged**